### PR TITLE
Integration tests

### PR DESCRIPTION
**Pull request by JoeARO**

### CODE REVIEW

**Comment by woodyblah** [comment]
Could you also add a test for fetching more than one component?

File: test/integration_test.exs [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+      assert conn.resp_body == expected_body
+    end
+  end
+
+  describe "when fetching the component receives a server error" do
+    test "it returns a 200, with the failing component status set as 500 and its original id still set" do
+      json_body = ~s({
+        "components": [{
+                        "id": "weather-front-page",
+                        "endpoint": "localhost:8082/non_existent_path",
```
Comment (ettomatic) [inline test/integration_test.exs]: this sounds like mocking a 404 to me

Comment (ettomatic) [reply test/integration_test.exs]: you should use: `get "/non_200_status/:code" do`

Comment (JoeARO) [reply test/integration_test.exs]: sure

File: test/integration_test.exs [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+      assert conn.resp_body == expected_body
+    end
+  end
+
+  describe "when fetching the component times out" do
+    test "it returns a 200, with the failing component status set as 408 and its original id still set" do
+      json_body = ~s({
+        "components": [{
+                        "id": "weather-front-page",
+                        "endpoint": "localhost:8082/non_200_status/408",
```
Comment (woodyblah) [inline test/integration_test.exs]: There is also `/timeout` which times out for real, but this works too.

File: test/integration_test.exs [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+              id: "news-front-page",
+              envelope: %{
+                head: [],
+                bodyLast: [],
+                bodyInline: "<DIV id=\"site-container\">"
+              }
+            },
+            %{
+              status: 200,
+              index: 0,
```
Comment (samfrench) [inline test/integration_test.exs]: Don't we expect this to be index 1?

Comment (woodyblah) [reply test/integration_test.exs]: There is a ticket for this. 

https://jira.dev.bbc.co.uk/browse/RESFRAME-2699

**Pull request merged**